Tests/ClientTests/Frontend/Browser/TabManagement/Tab+ChangeUserAgentTests.swift

```diff
@@ -0,0 +1,44 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+
+@testable import Client
+
+import XCTest
+
+class ChangeUserAgentTests: XCTestCase {
+    struct ChangeUserAgentsMockData {
+        let urlString: String
+        let userAgentWasChanged: Bool
+        let isPrivate: Bool
+        let expectContainsURL: Bool
+    }
+    // Create a mock file to use for testing
+    let testFile = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent("test-changed-ua-set-of-hosts.xcarchive")
+    override func setUp() {
+        // Ensure test file is empty before each test
+        try? Data().write(to: testFile)
+    }
+    func testUpdateAndContains() {
```

Comment:
nit, Could you add some spaces between functions? (in other places too)
```suggestion
    }

    func testUpdateAndContains() {
```

Reply:
swiftLinter was giving me the harassing me for adding spaces but I've fixed that:))

Reply:
Oh yeah we need a specific option toggled in Xcode, otherwise the spacing aren't set as needed and Swiftlint isn't happy with us 🙈